tests/index.js

```diff
@@ -170,35 +170,57 @@ test('sagaTestEngine throws under bad conditions', t => {
     () => sagaTestEngine(f, badMapping),
     'Env Mapping is missing a value for "key"')
 
+  // Bad mapping for saga that yields obj with anonymous function
+  function namedFunction() {}
+  const f2 = function*() {
+    yield { func: namedFunction }
+  }
+  const badMapping2 = [['no', 'functions']]
+  t.throws(
+    () => sagaTestEngine(f2, badMapping2),
+    `Env Mapping is missing a value for ${JSON.stringify({ "func": "[Function: namedFunction]" }, null, 2)}`
+  )
+
+  // Bad mapping for saga that yields object with anonymous function
+  var anonymousFunction = function() { return 'something'; }
+  const f3 = function*() {
+    yield { func: anonymousFunction }
+  }
+  const badMapping3 = [['also no', 'functions']]
+  t.throws(
+    () => sagaTestEngine(f3, badMapping3),
+    `Env Mapping is missing a value for ${JSON.stringify({ func: `[Function]: ${anonymousFunction.toString()}` }, null, 2)}`
+  )
```

Comment:
Looks like this fails on newer versions of node.
```bash
✔ redux-saga-test-engine (e8b2e13...) 🚀  nvm use 6.0.0
Now using node v6.0.0 (npm v3.8.6)
✔ redux-saga-test-engine (e8b2e13...) 🚀  yarn run test
yarn run v0.23.2
$ ava tests/

  10 passed

✨  Done in 0.68s.
✔ redux-saga-test-engine (e8b2e13...) 🚀  nvm use 7.0.0
Now using node v7.0.0 (npm v3.10.8)
✔ redux-saga-test-engine (e8b2e13...) 🚀  yarn run test
yarn run v0.23.2
$ ava tests/

  9 passed
  1 failed

  sagaTestEngine throws under bad conditions
  /Users/tbuckley/dnainfo/redux-saga-test-engine/tests/index.js:190

   189:   const badMapping3 = [['also no', 'functions']]
   190:   t.throws(
   191:     () => sagaTestEngine(f3, badMapping3),

  Env Mapping is missing a value for {
    "func": "anonymousFunction"
  }


error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

The issue seems to be that function expressions like `var anonymousFunction = function() { return '...' }` get assigned the variable name for their function name.